src/builder.rs

```diff
@@ -438,6 +441,19 @@ impl<Other: ExtendableThing, Status> ThingBuilder<Other, Status> {
             }
         };
 
+        let invalid_uri_variables = uri_variables
```

Comment:
`InteractionAffordance::uri_variables` needs a similar check. maybe factor it out to a function.

Reply:
Good catch, I did not realize it.